### DISCUSSION TITLE
Comparison Setup

### DISCUSSION THREAD

**Discussion by andy-81** [BottlecapDave/HomeAssistant-OctopusEnergy#941] [Q&A]
Hi All,

I am trying to set up the comparison feature in HA using this integration to see how much it would cost me on different tariffs but so far have been unable to set it up, I keep getting the error that the product or tariff code does not exist.

I am sure I have missed something but found myself going around in circles for the past few days and can’t quite figure out where I am going wrong.

I managed to get my tariff code from the API by logging in with the API and viewing my account details, and the comparison one I am trying to use the one I received again via API as documented in the how to guide, but I am still getting the error, can anyone help point out where I am going wrong.

attached is the screenshot I get with the two tariffs on.
![IMG_1712](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/16067130/9d7c9f71-4d11-427d-9590-79e191ef4fc4)

**Comment by BottlecapDave** [comment]
Hello. As per the [docs](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/setup/tariff_comparison/#tariff-code), you need to set the product code to the code of the product you want to target. The screenshot and based on your description, it sounds like you've set the `product code` to your current tariff not the `product code` of the tariff you wish to target. 

In your situation, you need to set the product code to `AGILE-24-04-03`, assuming you're trying to compare an agile tariff. This is unfortunately separate because there are some tairffs where the product code is not a sub string of the tariff code.

I'll try and update the docs to make this clearer.

**Reply by andy-81** [reply]
My god I am such an idiot.

I thought the first box was for the first tarrif and the second box was for the tarrif I wanted to compare against the first. 

Thank you so much for correcting me, I knew it was something I was doing wrong. 

Thanks again
